### PR TITLE
feat:

### PR DESCRIPTION
**Pull request by jmichelin**

### CODE REVIEW

File: generateTemplat.sh [hunk 1]
```diff
@@ -10,7 +10,10 @@ if [ "$1" != "" ]; then
     mkdir $1
     cd $1
     echo "{ \"lesson_title\": \"$1\" }" >> lesson_settings.json
+    echo "## $1" >> lesson.md
```
Comment (manzoid) [inline generateTemplat.sh]: this file has a typo in the name, also we're doing kebab-case.

File: single-function/add-two-numbers/02-impl-revision.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const addTwoNumbers = (l1, l2) => {
+  let carryover = 0,
+    dummyhead = { val: 0, next: null },
```
Comment (manzoid) [inline single-function/add-two-numbers/02-impl-revision.js]: let's put each declaration on its own line, with its keyword.

File: single-function/add-two-numbers/02-impl-revision.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    curr = dummyhead;
+
+  while (l1 !== null || l2 !== null) {
+    let list1Value = l1 !== null ? l1.val : 0;
```
Comment (manzoid) [inline single-function/add-two-numbers/02-impl-revision.js]: can these be `const` -- we should favor `const` wherever possible.

File: single-function/add-two-numbers/add-two-numbers.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const getTargetSumPaths = require('./02-impl-revision.js');
+
+describe('addTwoNumbers', () => {
+  it('will do the thing', () => {
```
Comment (manzoid) [inline single-function/add-two-numbers/add-two-numbers.test.js]: fix placeholder test name

File: single-function/add-two-numbers/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+## Prompt
+
+You are given two non-empty linked lists representing two non-negative integers. 
+The digits are stored in reverse order and each of their nodes contain a single digit. 
+Add the two numbers and return it as a linked list.
+
+You may assume the two numbers do not contain any leading zero, except the number 0 itself.
```
Comment (manzoid) [inline single-function/add-two-numbers/README.md]: > except the number 0 itself.
unless the entire number is 0.

(original wording is a bit ambiguous between a "number" and one "digit" of a number).

File: single-function/add-two-numbers/02-impl-revision.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+const addTwoNumbers = (l1, l2) => {
+  let carryover = 0,
+    dummyhead = { val: 0, next: null },
+    curr = dummyhead;
```
Comment (manzoid) [inline single-function/add-two-numbers/02-impl-revision.js]: dummyhead -> dummyHead

File: generateTemplat.sh (hunk 1 above)
Comment (jmichelin) [reply generateTemplat.sh]: we don't get to choose the case, its based on repl.it requirements.

File: single-function/add-two-numbers/02-impl-revision.js (hunk 2 above)
Comment (jmichelin) [reply single-function/add-two-numbers/02-impl-revision.js]: done

File: single-function/add-two-numbers/02-impl-revision.js (hunk 6 above)
Comment (jmichelin) [reply single-function/add-two-numbers/02-impl-revision.js]: done

File: single-function/add-two-numbers/02-impl-revision.js (hunk 3 above)
Comment (jmichelin) [reply single-function/add-two-numbers/02-impl-revision.js]: all possible variables set to const, only carryOver and curr needed to remain let

File: single-function/add-two-numbers/README.md (hunk 5 above)
Comment (jmichelin) [reply single-function/add-two-numbers/README.md]: ✔️

File: single-function/add-two-numbers/add-two-numbers.test.js (hunk 4 above)
Comment (jmichelin) [reply single-function/add-two-numbers/add-two-numbers.test.js]: ✔️

**Comment by jmichelin** [comment]
✔️

**Pull request merged**